ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -406,7 +411,11 @@ private[ember] object Parser {
         )
 
         resp <-
-          if (headerP.chunked) {
+          if (discardBody) {
+            (baseResp -> none[Array[Byte]].pure[F]).pure[F]
```

Comment:
Disclaimer, I haven't looked at the ember parser before.

What is this doing that is different from the line two below?

The return type is `F[(Response[F], Drain[F])]`. Presumably the thing on the right is what we are draining? Or is it the "drained bytes" 🤔 ?

My concern is whether or not
```scala
if (discardBody) {
  (baseResp -> none[Array[Byte]].pure[F]).pure[F]
```
is actually draining things from the socket or just saying "there wasn't anything on the socket".